multistep/commonsteps/step_http_server.go

```diff
@@ -111,13 +112,20 @@ func (s *StepHTTPServer) Run(ctx context.Context, state multistep.StateBag) mult
 		return multistep.ActionHalt
 	}
 
-	ui.Say(fmt.Sprintf("Starting HTTP server on port %d", s.l.Port))
+	httpAddr = s.HTTPAddress
+	httpAddrIface, present := state.GetOk("http_ip")
+	if present && httpAddrIface.(string) != "" {
+		httpAddr = httpAddrIface.(string)
+	}
+
+	ui.Say(fmt.Sprintf("Starting HTTP server on address %s and port %d", httpAddr, s.l.Port))
 
 	// Start the HTTP server and run it in the background
 	server := &http.Server{Addr: "", Handler: s.Handler()}
```

Comment:
I think we need to pass addr here:

```suggestion
	server := &http.Server{Addr: httpAddr, Handler: s.Handler()}
```

Reply:
Also addr is used in `Addr: s.HTTPAddress,`, I think we should that one there too